VotingApplication/VotingApplication.Web.Tests/E2E/BasicPollTests.cs

```diff
@@ -626,26 +626,81 @@ public void ChoiceAddingSubmission_AddsChoice()
             private static readonly string PollUrl = SiteBaseUri + "Poll/#/Vote/" + PollGuid;
-            private IWebDriver _driver;
 
-            [ClassInitialize]
-            public static void ClassInitialise(TestContext testContext)
+            [TestMethod, TestCategory("E2E")]
+            public void ElectionModePoll_DoesNotShowResultsButton()
             {
-                _context = new TestVotingContext();
+                using (IWebDriver driver = Driver)
+                {
+                    using (var context = new TestVotingContext())
+                    {
+                        Poll poll = CreatePoll(context, true);
+                        GoToUrl(driver, PollUrl);
 
-                List<Choice> testPollChoices = new List<Choice>() {
-                new Choice(){ Name = "Test Choice 1", Description = "Test Description 1" },
-                new Choice(){ Name = "Test Choice 2", 
-                              Description = "A very long test description 2 that should exceed the character limit for descriptions" }};
+                        IWebElement resultButton = FindElementById(driver, "results-button");
+
+                        Assert.IsFalse(resultButton.IsVisible());
+                    }
+                }
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void ElectionModePoll_WithVotes_NavigatesToResults()
+            {
+                using (IWebDriver driver = Driver)
+                {
+                    using (var context = new TestVotingContext())
+                    {
+                        Poll poll = CreatePoll(context, true);
+                        GoToUrl(driver, PollUrl);
+
+                        IReadOnlyCollection<IWebElement> voteButtons = FindElementsById(driver, "vote-button");
+                        voteButtons.First().Click();
+
+                        driver.Navigate().GoToUrl(PollUrl);
+
+                        WaitForPageChange();
+
+                        string expectedUrl = SiteBaseUri + "Poll/#/Results/" + PollGuid;
+
+                        Assert.IsTrue(driver.Url.StartsWith(expectedUrl));
+                    }
+                }
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void ElectionModeResults_WithoutVotes_NavigatesToVote()
+            {
+                using (IWebDriver driver = Driver)
+                {
+                    using (var context = new TestVotingContext())
+                    {
+                        Poll poll = CreatePoll(context, true);
+
+                        string resultsUrl = PollUrl.Replace("Vote", "Results");
+
+                        GoToUrl(driver, PollUrl);
+
+                        WaitForPageChange();
+
+                        Assert.AreEqual(PollUrl, driver.Url);
+                    }
+                }
+            }
+
+            public static Poll CreatePoll(TestVotingContext testContext, bool electionMode)
```

Comment:
Not so big a deal in a test, but I'm not a big fan of passing in bool flags to methods. I'd have gone with two separate methods, which could then be named more descriptively.